### ISSUE TITLE
Ratings/metadata edits not saved to XMP after fresh import of images

### ISSUE THREAD

**Issue by orthorhombic** [darktable-org/darktable#14913]
## Describe the bug

With a series of freshly imported dng files, when editing the metadata without editing the picture, an xmp file is not written. After an edit and the xmp file exists, updates to the metadata are immediately written to xmp after edits are made. 

This was observed when doing the following in the lighttable module:
- Using keyboard shortcuts to set rating or color label
- Using GUI buttons to set rating or color label
- Using the "metadata editor" to change fields such as description or creator

This was observed with the following setting applied:
Settings -> storage -> XMP -> write sidecar file for each image == "after edit"

I initially thought this may relate to the `dt_image_altered` function in 
https://github.com/darktable-org/darktable/blob/fcdce55aaf7de0dddccfcf4eee886518eee16807/src/common/image.c#L2736-L2743

However, quickly poking around in the database, it didn't seem like the hashes were changing with rating updates, but that may be intended. This would imply there is maybe another place I overlooked where metadata updates trigger xmp writing. If it is at all helpful, I've included the database queries I used to look around the sqlite library.

database queries used for troubleshooting:
```sql
select filename, history_hash.* from 
history_hash
inner join images on history_hash.imgid=images.id
where filename="my_file.dng"
```

```sql
SELECT filename, CASE
  WHEN basic_hash == current_hash THEN "DT_HISTORY_HASH_BASIC"
  WHEN auto_hash == current_hash THEN "DT_HISTORY_HASH_AUTO"
  WHEN (basic_hash IS NULL OR current_hash != basic_hash) AND
       (auto_hash IS NULL OR current_hash != auto_hash) THEN "DT_HISTORY_HASH_CURRENT"
  ELSE "DT_HISTORY_HASH_BASIC" END AS status
 FROM main.history_hash
inner join images on history_hash.imgid=images.id
where filename="my_file.dng"
```

## Commit

No response

## Where did you install darktable from?

darktable installed via flatpak

## darktable version

4.4.1

## What OS are you using?

Linux

## What is the version of your OS?

Ubuntu 22.04

## Describe your system?

darktable installed via flatpak
OS : Linux - kernel 6.2.0-25-generic
Linux - Distro : Ubuntu 22.04
Memory : 64gb
Graphics card : Nvidia RTX 3050

### Are you using OpenCL GPU in darktable?

Yes

### If yes, what is the GPU card and driver?

Nvidia RTX 3050, Driver Version: 535.54.03    CUDA Version: 12.2

**Comment by jenshannoschwalm** [comment]
I think a) your suspicion of `dt_image_altered()` is fully correct and b) that **is** the only place writing the sidecar. (leaving explicit writing alone)

The 'altered' check does only include steps done in darkroom development so if we would want to write sidecars in an 'only changed metadata situation' we would need to check for that too.

**Comment by orthorhombic** [comment]
Thanks for the sanity check! The bit that is most curious to me is that the rating/metadata will be written to disk every time it changes as long as there has been at least one darkroom edit. 

If I'm understanding, `dt_history_hash_get_status` is the function to understand. I'm not terribly familiar with the database, the hashes, or when they change. From your experience, is there any obvious solution or thing that could be easily used to trigger writing sidecars in an 'only changed metadata situation'?

https://github.com/darktable-org/darktable/blob/fcdce55aaf7de0dddccfcf4eee886518eee16807/src/common/history.c#L1739C1-L1767C2

**Comment by jenshannoschwalm** [comment]
The hash is calculated just from the parameters of modules. So no Metadata included.

**Comment by fabiosirna** [comment]
@orthorhombic @jenshannoschwalm Today I had to deal with the same problem. I cleaned up my list of tags and metadata and tried to get my photos in order. Those that had been edited in the past have the updated XMP file. Those that were never edited keep the tags/metadata in darktable (in the db I guess) but the XMP file is not written. 

Although this is not a bug that prevents the use of darktable, I believe it is a problem for everyone like me who uses darktable for editing in conjunction with digikam for DAM.

From a user perspective, assigning a metadata or tag is still non-destructive editing of the image and therefore it would be correct to write the XMP file on demand (settings after edit).
